src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -531,18 +531,37 @@ internal static void ImportGlobalSymbols(this FileContentManager file, Compilati
         /// If the set of required functors is unspecified or null, then the functors to support are determined by the parent scope.
         /// </summary>
         private static QsScope BuildScope(
-            IReadOnlyList<FragmentTree.TreeNode> nodeContent,
+            IEnumerator<FragmentTree.TreeNode> nodes,
```

Comment:
What was the motivation to change this from `IReadOnlyList` to `IEnumerator`?

Reply:
When building an implicit scope, the statement nodes for the scope body re-use the same enumerator from the parent scope. Using an `IReadOnlyList` would require consuming the rest of the enumerator all at once and building a new list, which is unnecessary because `BuildScope` simply enumerates that list from beginning to end.